--- a/net/neoforged/neoforge/network/configuration/SyncConfig.java
+++ b/net/neoforged/neoforge/network/configuration/SyncConfig.java
@@ -7,9 +7,8 @@
 
 import java.util.function.Consumer;
 import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
-import net.minecraft.network.protocol.configuration.ServerConfigurationPacketListener;
 import net.minecraft.resources.Identifier;
-import net.neoforged.neoforge.common.NeoForgeMod;
+import net.minecraft.server.network.ServerConfigurationPacketListenerImpl;
 import net.neoforged.neoforge.network.ConfigSync;
 import org.jetbrains.annotations.ApiStatus;
 
@@ -19,14 +18,14 @@
  * @param listener the listener to indicate to that the task is complete
  */
 @ApiStatus.Internal
-public record SyncConfig(ServerConfigurationPacketListener listener) implements ICustomConfigurationTask {
-    private static final Identifier ID = Identifier.fromNamespaceAndPath(NeoForgeMod.MOD_ID, "sync_config");
-    public static Type TYPE = new Type(ID);
+public record SyncConfig(ServerConfigurationPacketListenerImpl listener) implements ICustomConfigurationTask {
+    private static final Identifier ID = fuzs.forgeconfigapiport.impl.ForgeConfigAPIPort.id("sync_config");
+    public static Type TYPE = new Type(ID.toString());
 
     @Override
     public void run(Consumer<CustomPacketPayload> sender) {
         ConfigSync.syncAllConfigs(listener);
-        listener().finishCurrentTask(type());
+        listener().completeTask(type());
     }
 
     @Override
