--- a/net/neoforged/neoforge/common/ModConfigSpec.java
+++ b/net/neoforged/neoforge/common/ModConfigSpec.java
@@ -39,10 +39,8 @@
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
-import net.neoforged.fml.Logging;
+import fuzs.forgeconfigapiport.impl.core.Logging;
 import net.neoforged.fml.config.IConfigSpec;
-import net.neoforged.fml.config.ModConfig;
-import net.neoforged.fml.loading.FMLEnvironment;
 import org.apache.commons.lang3.tuple.Pair;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -125,6 +123,7 @@
         this.afterReload();
     }
 
+    /*
     @Override
     public void validateSpec(ModConfig config) {
         forEachValue(getValues().valueMap().values(), configValue -> {
@@ -143,6 +142,7 @@
             }
         });
     }
+    */
 
     public boolean isLoaded() {
         return loadedConfig != null;
@@ -925,7 +925,7 @@
 
         public String buildComment(final List<String> path) {
             if (comment.stream().allMatch(String::isBlank)) {
-                if (FMLEnvironment.isProduction())
+                if (!fuzs.forgeconfigapiport.impl.services.CommonAbstractions.INSTANCE.isDevelopmentEnvironment())
                     LOGGER.warn(Logging.CORE, "Detected a comment that is all whitespace for config option {}, which causes obscure bugs in NeoForge's config system and will cause a crash in the future. Please report this to the mod author.",
                             DOT_JOINER.join(path));
                 else
@@ -1399,8 +1399,9 @@
          * <p>
          * Cannot be used for {@linkplain ModConfig.Type#SERVER server configs}.
          */
-        GAME(ModConfig.Type.SERVER);
+        GAME;
 
+        /*
         private final Set<ModConfig.Type> invalidTypes;
 
         RestartType(ModConfig.Type... invalidTypes) {
@@ -1411,6 +1412,7 @@
         private boolean isValid(ModConfig.Type type) {
             return !invalidTypes.contains(type);
         }
+        */
 
         public RestartType with(RestartType other) {
             return other == NONE ? this : (other == GAME || this == GAME) ? GAME : WORLD;
