--- a/net/neoforged/neoforge/common/ModConfigSpec.java
+++ b/net/neoforged/neoforge/common/ModConfigSpec.java
@@ -39,10 +39,7 @@
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
-import net.neoforged.fml.Logging;
 import net.neoforged.fml.config.IConfigSpec;
-import net.neoforged.fml.config.ModConfig;
-import net.neoforged.fml.loading.FMLEnvironment;
 import org.apache.commons.lang3.tuple.Pair;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -115,16 +112,17 @@
         if (config != null && !isCorrect(config.config())) {
             // Correct in case the config did not get corrected before this function was called.
             // This should not happen under normal circumstances, hence the warning.
-            LOGGER.warn(Logging.CORE, "Configuration {} is not correct. Correcting", config);
+            LOGGER.warn("Configuration {} is not correct. Correcting", config);
             correct(config.config(),
-                    (action, path, incorrectValue, correctedValue) -> LOGGER.warn(Logging.CORE, "Incorrect key {} was corrected from {} to its default, {}. {}", DOT_JOINER.join(path), incorrectValue, correctedValue, incorrectValue == correctedValue ? "This seems to be an error." : ""),
-                    (action, path, incorrectValue, correctedValue) -> LOGGER.debug(Logging.CORE, "The comment on key {} does not match the spec. This may create a backup.", DOT_JOINER.join(path)));
+                    (action, path, incorrectValue, correctedValue) -> LOGGER.warn("Incorrect key {} was corrected from {} to its default, {}. {}", DOT_JOINER.join(path), incorrectValue, correctedValue, incorrectValue == correctedValue ? "This seems to be an error." : ""),
+                    (action, path, incorrectValue, correctedValue) -> LOGGER.debug("The comment on key {} does not match the spec. This may create a backup.", DOT_JOINER.join(path)));
 
             config.save();
         }
         this.afterReload();
     }
 
+    /*
     @Override
     public void validateSpec(ModConfig config) {
         forEachValue(getValues().valueMap().values(), configValue -> {
@@ -143,6 +141,7 @@
             }
         });
     }
+    */
 
     public boolean isLoaded() {
         return loadedConfig != null;
@@ -624,13 +623,13 @@
                 @Override
                 public Object correct(Object value) {
                     if (!(value instanceof List) || (getSizeRange() != null && !getSizeRange().test(((List<?>) value).size()))) {
-                        LOGGER.debug(Logging.CORE, "List on key {} is deemed to need correction, as it is null, not a list, or the wrong size.", path.getLast());
+                        LOGGER.debug("List on key {} is deemed to need correction, as it is null, not a list, or the wrong size.", path.getLast());
                         return getDefault();
                     }
                     List<?> list = Lists.newArrayList((List<?>) value);
                     list.removeIf(elementValidator.negate());
                     if (list.isEmpty()) {
-                        LOGGER.debug(Logging.CORE, "List on key {} is deemed to need correction. It failed validation.", path.getLast());
+                        LOGGER.debug("List on key {} is deemed to need correction. It failed validation.", path.getLast());
                         return getDefault();
                     }
                     return list;
@@ -925,8 +924,8 @@
 
         public String buildComment(final List<String> path) {
             if (comment.stream().allMatch(String::isBlank)) {
-                if (FMLEnvironment.isProduction())
-                    LOGGER.warn(Logging.CORE, "Detected a comment that is all whitespace for config option {}, which causes obscure bugs in NeoForge's config system and will cause a crash in the future. Please report this to the mod author.",
+                if (!fuzs.forgeconfigapiport.impl.services.CommonAbstractions.INSTANCE.isDevelopmentEnvironment())
+                    LOGGER.warn("Detected a comment that is all whitespace for config option {}, which causes obscure bugs in NeoForge's config system and will cause a crash in the future. Please report this to the mod author.",
                             DOT_JOINER.join(path));
                 else
                     throw new IllegalStateException("Can not build comment for config option " + DOT_JOINER.join(path) + " as it comprises entirely of blank lines/whitespace. This is not allowed as it causes a \"constantly correcting config\" bug with NightConfig in NeoForge's config system.");
@@ -1037,7 +1036,7 @@
                 Number n = (Number) t;
                 boolean result = ((Number) min).doubleValue() <= n.doubleValue() && n.doubleValue() <= ((Number) max).doubleValue();
                 if (!result) {
-                    LOGGER.debug(Logging.CORE, "Range value {} is not within its bounds {}-{}", n.doubleValue(), ((Number) min).doubleValue(), ((Number) max).doubleValue());
+                    LOGGER.debug("Range value {} is not within its bounds {}-{}", n.doubleValue(), ((Number) min).doubleValue(), ((Number) max).doubleValue());
                 }
                 return result;
             }
@@ -1046,7 +1045,7 @@
 
             boolean result = c.compareTo(min) >= 0 && c.compareTo(max) <= 0;
             if (!result) {
-                LOGGER.debug(Logging.CORE, "Range value {} is not within its bounds {}-{}", c, min, max);
+                LOGGER.debug("Range value {} is not within its bounds {}-{}", c, min, max);
             }
             return result;
         }
@@ -1399,8 +1398,9 @@
          * <p>
          * Cannot be used for {@linkplain ModConfig.Type#SERVER server configs}.
          */
-        GAME(ModConfig.Type.SERVER);
+        GAME;
 
+        /*
         private final Set<ModConfig.Type> invalidTypes;
 
         RestartType(ModConfig.Type... invalidTypes) {
@@ -1411,6 +1411,7 @@
         private boolean isValid(ModConfig.Type type) {
             return !invalidTypes.contains(type);
         }
+        */
 
         public RestartType with(RestartType other) {
             return other == NONE ? this : (other == GAME || this == GAME) ? GAME : WORLD;
